persistence/src/main/scala/zio/actors/persistence/EventSource.scala

```diff
@@ -0,0 +1,126 @@
+package zio.actors.persistence
+
+import java.io.File
+
+import zio.actors.{ Actor, Context, Supervisor }
+import zio.{ IO, Queue, RIO, Ref, Task, ZIO }
+import zio.actors.Actor._
+import zio.actors.persistence.Utils.JournalPlugin
+import zio.actors.persistence.journal.{ InMemJournal, JDBCJournal, Journal }
+
+/**
+ *
+ * Each message can result in either an event that will be persisted or idempotent action.
+ * Changing the actor's state can only occur via `Persist` event.
+ *
+ * @tparam Ev events that will be persisted
+ */
+sealed trait Command[+Ev]
+case class Persist[+Ev](event: Ev) extends Command[Ev]
+case object Ignore                 extends Command[Nothing]
+object Command {
+  def persist[Ev](event: Ev): Persist[Ev] = Persist(event)
+
+  def ignore: Ignore.type = Ignore
+}
+
+/**
+ *
+ * Description of event sources actor's behavior
+ *
+ * @param persistenceId unique id used in a datastore for identifying the entity
+ * @tparam S state type
+ * @tparam E type of errors that might be produced while processing messages
+ * @tparam F type of messages that actor receives
+ * @tparam Ev events that will be persisted
+ */
+abstract class EventSourcedStateful[R, S, +E <: Throwable, -F[+_], Ev](persistenceId: String)
+    extends AbstractStateful[R, S, E, F] {
+
+  def receive[A](state: S, msg: F[A], context: Context): ZIO[R, E, (Command[Ev], S => A)]
+
+  def sourceEvent(state: S, event: Ev): S
+
+  /* INTERNAL API */
+
+  override final def constructActor(
```

Comment:
Similar to the one before, this method doesn't belong here but in its companion. Also, adopt ZIO's naming convention, rename `convertActor` to `makeActor`.